python/cuml/test/test_fil.py

```diff
@@ -109,25 +108,76 @@ def test_fil(n_rows, n_columns, n_info, num_round, tmp_path):
     # split validation data
     X_validation, y_validation = X[train_index:, :], y[train_index:]
 
+    model_path = os.path.join(tmp_path, 'xgb_class.model')
 
-    model_path = os.path.join(tmp_path, 'xgb.model')
-
-    bst = _build_and_save_xgboost(model_path, X_train, y_train)
+    bst = _build_and_save_xgboost(model_path, X_train, y_train, classification)
 
     dvalidation = xgb.DMatrix(X_validation, label=y_validation)
     xgb_preds = bst.predict(dvalidation)
 
     xgb_acc = accuracy_score(y_validation, xgb_preds > 0.5)
 
     print("Reading the saved xgb model")
+
     fm = FIL.from_treelite_file(model_path,
                                 algo=0,
                                 output_class=True,
                                 threshold=0.50)
-    preds_1 = np.asarray(fm.predict(X_validation))
-    fil_acc = accuracy_score(y_validation, preds_1)
+    fil_preds = np.asarray(fm.predict(X_validation))
+    fil_acc = accuracy_score(y_validation, fil_preds)
 
     print("XGB accuracy = ", xgb_acc, " FIL accuracy: ", fil_acc)
     assert fil_acc == pytest.approx(xgb_acc, 0.01)
     assert fil_acc > 0.80
 
+
+@pytest.mark.parametrize('n_rows', [unit_param(100), quality_param(1000),
+                         stress_param(500000)])
+@pytest.mark.parametrize('n_columns', [unit_param(30), quality_param(100),
+                         stress_param(1000)])
+@pytest.mark.parametrize('num_round', [unit_param(20), quality_param(50),
+                         stress_param(90)])
+@pytest.mark.skipif(has_xgboost() is False, reason="need to install xgboost")
+def test_fil_reg(n_rows, n_columns, num_round, tmp_path):
+    # settings
+    classification = False  # change this to false to use regression
+    n_rows = n_rows  # we'll use 1 millions rows
+    n_columns = n_columns
+    random_state = np.random.RandomState(43210)
+
+    dataset = simulate_data(n_rows, n_columns,
+                            random_state=random_state,
+                            classification=classification)
+    # identify shape and indices
+    n_rows, n_columns = dataset.shape
+    train_size = 0.80
+    train_index = int(n_rows * train_size)
+
+    # split X, y
+    X, y = dataset[:, 1:], dataset[:, 0]
+    del dataset
+
+    # split train data
+    X_train, y_train = X[:train_index, :], y[:train_index]
```

Comment:
Why not use model_selection.train_test_split? I think that's pretty handy

Reply:
will make the change